src/components-examples/material/checkbox/checkbox-overview/checkbox-overview-example.ts

```diff
@@ -8,4 +16,33 @@ import {Component} from '@angular/core';
   templateUrl: 'checkbox-overview-example.html',
   styleUrls: ['checkbox-overview-example.css'],
 })
-export class CheckboxOverviewExample {}
+export class CheckboxOverviewExample {
+  task: Task = {
+    name: 'Indeterminate',
+    completed: false,
+    color: 'primary',
+    subtasks: [
+      {name: 'Primary', completed: false, color: 'primary'},
+      {name: 'Accent', completed: false, color: 'accent'},
+      {name: 'Warn', completed: false, color: 'warn'}
+    ]
+  };
+
+  allComplete(): boolean {
+    return this.task.subtasks != null && this.task.subtasks.every(t => t.completed);
+  }
+
+  someComplete(): boolean {
+    if (this.task.subtasks == null) {
+      return false;
+    }
+    return this.task.subtasks.filter(t => t.completed).length > 0 && !this.allComplete();
```

Comment:
This is definitely the cleaner way to do it, but I'm not sure whether it's the best in terms of performance, because this method and `allComplete` will be invoked on every change detection run. It might be better to have this in a boolean property and recalculate it once when a value has changed.

Reply:
Updated to make `allComplete` a boolean, and changed the method to `updateAllComplete`, which sets `allComplete` when subtasks change and when calling `setAll`. Was that along the lines of what you were thinking?